go-selinux/selinux_linux.go

```diff
@@ -406,7 +406,19 @@ func SocketLabel() (string, error) {
 // SetKeyLabel takes a process label and tells the kernel to assign the
 // label to the next kernel keyring that gets created
 func SetKeyLabel(label string) error {
-	return writeCon("/proc/self/attr/keycreate", label)
+	if label == "" && GetEnabled() {
+		if _, err := KeyLabel(); err == io.EOF {
```

Comment:
Why not check ENoExists here?

Reply:
Thanks, Because we add it after write action. Anyway, I think we can add. I will add it later.

Reply:
And the ENoExists check after `writeCon` should also be remained.

Reply:
Thank you for your review, it will reduce one file write call, does these code change make sense?

Reply:
It will still fail on Permission Denied.  If the user tries to write to a KeyLabel they are allowed to read but not allowed to write.
I am not even sure what the io.EOF does?

Reply:
I think it does make sense that Permission Denied err when write to keycreate should be throwed to runc.

Reply:
> I am not even sure what the io.EOF does?

I think it is what is the older kernel really mean! It also be very very strange to me, but it really happens. I think this is a selinux bug in this old kernel.

Reply:
What OS and What Kernel?

Reply:
> What OS and What Kernel?

I think I have provided it in PR description.
In some old kernels, for example, 3.10.0-862.el7.x86_64 in RedHat Enterprise Linux 7.5.
With selinux enabled.